keps/sig-cli/kustomize-secret-generator-plugins.md

```diff
@@ -205,52 +266,114 @@ are out of scope for this KEP, however this KEP's
+with the same version of Go, with the same libs, on the same
+machine (see, e.g. [this golang issue](https://github.com/golang/go/issues/18827)).
+
+Therefore, the Go developer who elects to write their
+key:value generator as a goplugin is obligated to compile a
+new kustomize binary as well as the plugin.
+
+For this developer, this extra compilation step will be far
+easier than forking kustomize and modifying the code
+directly to introduce all the piping one would need to
+express a new key:value secret generation style in a
+kustomization file and integrate it with existing methods.
+
+The risk to not compiling, and, say using an `.so` that was
+compiled long ago and far away is that the program will
+panic when the `build` command is executed.
+
+##### mitigation
+
+To remind users of this risk, a special runtime
+flag will be required; see _graduation criteria_ below.
+
+##### No current support for Windows
 
-#### No current windows support for golang plugins.
-This may be implemented by the Go team later in which
-case it will just work.
+This is mentioned in the
+[plugin package overview section](https://golang.org/pkg/plugin).
 
-Also, as noted above, someone can write an exec style plugin
+##### mitigation
+
+This KEP offers a framework that dramatically
+lowers the barrier to someone writing an exec style plugin
 which will work on Windows.
 
+Of course, Windows users who want to support
+goplugins could contribute to the golang project.
+
 #### General symbol execution from the plugin
-A Go plugin will be cast to a particular hardcoded interface, e.g.
+
+##### mitigation
+
+A Go plugin, when loaded,
+will be cast to a particular hardcoded interface, e.g.
 
 ```
 type KvSourcePlugin {
-	Get []kvPairs
+	Get []kv.Pair
 }
 ```
 and accessed exclusively through that interface.
 
-#### Existing KV generators continue as an option
-We leave in place the existing three mechanisms
-(_literals_, _env_ and _files_) for generating KV pairs, but
-additionally allow these mechanisms to be expressed as
-`builtin` plugins (see example above).
+### Two means to specify legacy KV generation
+
+The existing three mechanisms and syntax for generating KV
+pairs (_literals_, _env_ and _files_) remain active, but
+additionally the work for this KEP will allow these
+mechanisms to be expressed as `builtin` plugins (see example
+above).
+
+__mitigation:__
 
-If plugins - both Go and other styles - prove unpopular
-or problematic, we can remove them per API change
+If plugins - both goplugins and other styles - prove
+unpopular or problematic, we can remove them per API change
 policies.
 
-If they do prove popular/useful, we can remove deprecate
-the legacy form for (_literals_, _env_ and _files_),
-and help people convert to the new "builtin" form.
+If they do prove popular/useful, we can deprecate the legacy
+form for (_literals_, _env_ and _files_), and help people
+convert to the new `builtin` form to access these KV sources.
 
 ## Graduation Criteria
```

Comment:
We should say that this is alpha, and will be deprecated and removed if it does not have an active maintainer.  We should have a separate owners file for the package that defines who the owner is so that it is explicitly tracked separately from everything else.

Reply:
added